### DISCUSSION TITLE
CareLink and microbolus time

### DISCUSSION THREAD

**Discussion by Navid200** [NightscoutFoundation/xDrip#2881] [Q&A]
@benceszasz 
Hi,

Someone has posted a question on facebook showing this:
![347265589_6368789029876913_1714165208087864325_n](https://github.com/NightscoutFoundation/xDrip/assets/51497406/f7d52ea4-19c1-4959-a42e-24ea81867e99)

xDrip automatically creates a reminder when a treatment is entered in the future.
Many disable the notification so that it would make no sound.  But, I don't believe that's the correct approach.
If someone intentionally enters a treatment in the future, they should get an alert when the time comes.  If they have disabled the sound, they may never know.

Why is the micobolus entered with a time parameter in the future?

**Comment by benceszasz** [comment]
@Navid200 everything (manual bolus, food, autocorrection, SG, notification, etc.) is added using its own timestamp from CareLink. There is a time lag (20-60 seconds) before SG data is available from CareLink. Microboluses are given automatically by the 780G based on the last SG value, so these entries are the latest ones, but still, these are events from the past. I ran a test with a few different 780Gs users and didn't get a treatment reminder. After that I have changed the follower device's clock to a few minutes in the past and after a few hours I have finally received one treatment reminder for a meal entry, which was added 27 seconds after the last SG. Not sure, that the clocks are causing the error (either the pump's or the phone's), but this was the only way I was able to reproduce this error. Based on my tests I am curious: how many is that "Many"?

**Reply by Navid200** [reply]
I suspect the ones who see the treatment reminder have a time differences between the master and follower phones.  And I have asked them to confirm.

Thanks
